src/main/java/ratpack/zipkin/internal/RatpackServerClientLocalSpanState.java

```diff
@@ -37,24 +37,29 @@
   private final MDCProxy mdc;
   private final Endpoint endpoint;
 
-  public RatpackServerClientLocalSpanState(final String serviceName, int ip, int port) {
-    this(serviceName, ip, port, Execution::current, new DefaultMDCProxyImpl());
-  }
-
-  RatpackServerClientLocalSpanState(final String serviceName,
-                                    int ip,
-                                    int port,
-                                    final Supplier<MutableRegistry> registry,
-                                    final MDCProxy mdc) {
+  public RatpackServerClientLocalSpanState(final Endpoint endpoint,
+                                           final Supplier<MutableRegistry> registry,
+                                           final MDCProxy mdc) {
     this.registry = registry;
-    this.endpoint = Endpoint.builder()
-                            .serviceName(serviceName)
-                            .ipv4(ip)
-                            .port(port)
-                            .build();
+    this.endpoint = endpoint;
     this.mdc = mdc;
   }
 
+  public RatpackServerClientLocalSpanState(final Endpoint endpoint) {
```

Comment:
Are the other 2 public constructors defined here ever even used?  If not we should consider removing them or making them non public if possible.